static/app/views/performance/newTraceDetails/traceDrawer/traceDrawer.tsx

```diff
@@ -701,24 +701,22 @@ function TabPinButton(props: {
   onClick?: (e: React.MouseEvent<HTMLElement>) => void;
 }) {
   return (
-    <PinButton
-      size="zero"
-      data-test-id="trace-drawer-tab-pin-button"
-      onClick={props.onClick}
-    >
+    <PinButton data-test-id="trace-drawer-tab-pin-button" onClick={props.onClick}>
```

Comment:
Isn’t this more like a Button with `priority="transparent"` where the `StyledIconPin` is passed to the `icon` slot ?

- we’re also missing a `label` here - the Button would enforce that
- the `data-test-id` is also a bit of a red flag, as buttons are interactive (likely goes together with the missing label)

Reply:
The one issue we have with transparent variant is that it is only supported in chonk, which means we need to @ts-expect-error until the legacy buttons are removed. I've first avoided that, but I think it makes sense. The violations will surface once the old button is removed. 

I am also tempted to backport transparent variant to the current buttons set, as it seems like it could be useful here too (note that I've had to remove box shadow and border from the button)

Agree about the label, and I've moved the icon to use the icon slot.